client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -8,10 +9,97 @@
+            for example "ProjectWorkflow.Default"
+
+    Returns:
+        Any: Substance Python API object of the project attribute.
+
+    Raises:
+        ValueError: If attribute does not exist on the
+            `substance_painter.project` python api.
+    """
+    root = substance_painter.project
+    for attr in substance_attr.split("."):
+        root = getattr(root, attr, None)
+        if root is None:
+            raise ValueError(
+                "Substance Painter project attribute"
+                f" does not exist: {substance_attr}")
+
+    return root
+
+
+def get_template_by_name(name: str, templates: list[dict]) -> dict:
+    return next(
+        template for template in templates
+        if template["name"] == name
+    )
+
+
+class SubstanceProjectConfigurationWindow(QtWidgets.QDialog):
+    """The pop-up dialog allows users to choose material
+    duplicate options for importing Max objects when updating
+    or switching assets.
+    """
+    def __init__(self, project_templates):
+        super(SubstanceProjectConfigurationWindow, self).__init__()
+        self.setWindowFlags(self.windowFlags() | QtCore.Qt.FramelessWindowHint)
+
+        self.import_cameras = False
+        self.preserve_strokes = False
+        self.template_name = None
+        self.project_templates = project_templates
+
+        self.widgets = {
+            "label": QtWidgets.QLabel("Project Configuration"),
+            "template_options": QtWidgets.QComboBox(),
+            "import_cameras": QtWidgets.QCheckBox("Import Cameras"),
+            "preserve_strokes": QtWidgets.QCheckBox("Preserve Strokes"),
+            "clickbox": QtWidgets.QWidget(),
+            "combobox": QtWidgets.QWidget(),
+            "okButton": QtWidgets.QPushButton("Ok"),
+        }
+        for template in project_templates:
+            self.widgets["template_options"].addItem(template)
+
+        # Build clickboxes
+        layout = QtWidgets.QHBoxLayout(self.widgets["clickbox"])
+        layout.addWidget(self.widgets["import_cameras"])
+        layout.addWidget(self.widgets["preserve_strokes"])
+        # Build buttons.
+        layout = QtWidgets.QHBoxLayout(self.widgets["combobox"])
+        layout.addWidget(self.widgets["template_options"])
+        layout.addWidget(self.widgets["okButton"])
+        # Build layout.
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(self.widgets["label"])
+        layout.addWidget(self.widgets["clickbox"])
+        layout.addWidget(self.widgets["combobox"])
+
+        self.widgets["okButton"].pressed.connect(self.on_ok_pressed)
+
+    def on_ok_pressed(self):
+        if self.widgets["import_cameras"].isChecked():
+            self.import_cameras = True
+        if self.widgets["preserve_strokes"].isChecked():
+            self.preserve_strokes = True
+        self.template_name = (
+            self.widgets["template_options"].currentText()
+        )
+        self.close()
```

Comment:
I wonder if it's better to make a `staticmethod` or `classmethod` that just gets the templates and visualizes them and at the end returns a dict of the resulting values. I think the behavior would become way clearer.

Along the lines of:
```python
    @classmethod
    def prompt(cls, templates):
        dialog = cls(templates)
        dialog.exec_()
        return dialog.get_result()
```

So that e.g. when wanting to ask the user for values you can just do:
```python
result = SubstanceProjectConfigurationWindow.prompt(templates)
```

---

By the way, what happens if the user 'cancels' the dialog? I believe currently it will just continue the logic with the default values, not the picked values at that point in time.

Even better maybe if it would stop the full operation completely?